nixos/modules/system/activation/activation-script.nix

```diff
@@ -158,8 +158,11 @@ in
         ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.devSize}" none /dev
         ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.devShmSize}" none /dev/shm
         ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.runSize}" none /run
+      '' + optionalString config.boot.tmpOnTmpfs ''
+        if [ mountpoint -q /tmp ]; then
```

Comment:
boot.tmpOnTmpfs is only going to apply /on boot/. The activation-script shouldn't attempt to remount if you set boot.tmpOnTmpfs to true without rebooting.